pkg/resources/hetzner/resources.go

```diff
@@ -0,0 +1,293 @@
+	}
+
+	for _, firewall := range firewalls {
+		resourceTracker := &resources.Resource{
+			Name:    firewall.Name,
+			ID:      strconv.Itoa(firewall.ID),
+			Type:    resourceTypeFirewall,
+			Deleter: deleteFirewall,
+			Obj:     firewall,
+		}
+
+		resourceTrackers = append(resourceTrackers, resourceTracker)
+	}
+
+	return resourceTrackers, nil
+}
+
+func listLoadBalancers(cloud fi.Cloud, clusterName string) ([]*resources.Resource, error) {
+	c := cloud.(hetzner.HetznerCloud)
+	var resourceTrackers []*resources.Resource
+
+	loadBalancers, err := c.GetLoadBalancers(clusterName)
+	if err != nil {
+		return nil, fmt.Errorf("failed to list load balancers: %v", err)
+	}
+
+	for _, loadBalancer := range loadBalancers {
+		resourceTracker := &resources.Resource{
+			Name:    loadBalancer.Name,
+			ID:      strconv.Itoa(loadBalancer.ID),
+			Type:    resourceTypeLoadBalancer,
+			Deleter: deleteLoadBalancer,
+			Obj:     loadBalancer,
+		}
+
+		resourceTrackers = append(resourceTrackers, resourceTracker)
+	}
+
+	return resourceTrackers, nil
+}
+
+func listServers(cloud fi.Cloud, clusterName string) ([]*resources.Resource, error) {
+	c := cloud.(hetzner.HetznerCloud)
+	var resourceTrackers []*resources.Resource
+
+	servers, err := c.GetServers(clusterName)
+	if err != nil {
+		return nil, fmt.Errorf("failed to list servers: %v", err)
+	}
+
+	for _, server := range servers {
+		resourceTracker := &resources.Resource{
+			Name:    server.Name,
+			ID:      strconv.Itoa(server.ID),
+			Type:    resourceTypeServer,
+			Deleter: deleteServer,
+			Obj:     server,
+		}
+
+		resourceTrackers = append(resourceTrackers, resourceTracker)
+	}
+
+	return resourceTrackers, nil
+}
+
+func listVolumes(cloud fi.Cloud, clusterName string) ([]*resources.Resource, error) {
+	c := cloud.(hetzner.HetznerCloud)
+	var resourceTrackers []*resources.Resource
+
+	volumes, err := c.GetVolumes(clusterName)
+	if err != nil {
+		return nil, fmt.Errorf("failed to list volumes: %v", err)
+	}
+
+	for _, volume := range volumes {
+		resourceTracker := &resources.Resource{
+			Name:    volume.Name,
+			ID:      strconv.Itoa(volume.ID),
+			Type:    resourceTypeVolume,
+			Deleter: deleteVolume,
+			Obj:     volume,
+		}
+
+		resourceTrackers = append(resourceTrackers, resourceTracker)
+	}
+
+	return resourceTrackers, nil
+}
+
+func deleteSSHKey(cloud fi.Cloud, r *resources.Resource) error {
+	klog.Infof("Deleting SSH Key: %s(%s)", r.Name, r.ID)
+
+	c := cloud.(hetzner.HetznerCloud)
+	client := c.SSHKeyClient()
+	sshKey := r.Obj.(*hcloud.SSHKey)
+	_, err := client.Delete(context.TODO(), sshKey)
+	if err != nil {
+		return fmt.Errorf("failed to delete ssh key %s(%s): %w", r.Name, r.ID, err)
+	}
+
+	return nil
+}
+
+func deleteNetwork(cloud fi.Cloud, r *resources.Resource) error {
+	klog.Infof("Deleting Network: %s(%s)", r.Name, r.ID)
+
+	c := cloud.(hetzner.HetznerCloud)
+	client := c.NetworkClient()
+	network := r.Obj.(*hcloud.Network)
+	_, err := client.Delete(context.TODO(), network)
```

Comment:
This makes me want to go and add ctx to all our delete methods :-)  I'm not going to force the merge-conflict on you though!